### PR TITLE
Attach Camera To Any Moveable  / Mesh.

### PR DESCRIPTION
**Pull request by Nemoel-Tomo**
Moveable:AttachCamera( short Meshid1, short MeshID2)
Attach the game camera to any moveable. set number of mesh (Mesh1ID9 of the object to attach camera to and number of mesh (Mesh2ID) to attach the camera target to another mesh of the object.

Misc function:
 ResetObjCamera()
to reset the camera back to normal

### CODE REVIEW

File: TombEngine/Scripting/Internal/TEN/Objects/Moveable/MoveableObject.cpp [hunk 1]
```diff
@@ -1018,3 +1022,11 @@ bool Moveable::MeshExists(int index) const
 
 	return true;
 }
+
+void  Moveable::AttachCamera(short CamMeshId, short TargetMeshId)
```
Comment (Sezzary) [inline TombEngine/Scripting/Internal/TEN/Objects/Moveable/MoveableObject.cpp:1026]: Parameter names should be in camelCase (both definition and declaration).

File: TombEngine/Game/camera.cpp [hunk 2]
```diff
@@ -323,6 +332,131 @@ void MoveCamera(GameVector* ideal, int speed)
 	}
 }
 
+
+void ObjCamera(ItemInfo* CamSlotId, int CamMeshID, ItemInfo* TargetItem, int TargetMeshID, bool cond)
+{
+	GameVector from, to;
```
Comment (Sezzary) [inline TombEngine/Game/camera.cpp:338]: No need to declare local variables at the start of the function. I believe that's a practice from the 80s when compilers needed it, but these days it adds a lot of clutter. Please initialise local variables on the first instance of use, like this:
`auto from = GameVector(dest, CamSlotId->RoomNumber);`

File: TombEngine/Game/camera.cpp [hunk 3]
```diff
@@ -323,6 +332,131 @@ void MoveCamera(GameVector* ideal, int speed)
...
+
+	//CamSlotId and TargetItem stay the same object until I know how to expand TargetItem to another object
```
Comment (Sezzary) [inline TombEngine/Game/camera.cpp:340]: Have a space after `//` and finish the sentence with a `.` when adding comments. Keeps things tidy. :)

File: TombEngine/Game/camera.cpp [hunk 4]
```diff
@@ -323,6 +332,131 @@ void MoveCamera(GameVector* ideal, int speed)
...
+		
+	int	speed = 1;
+		//Get mesh1 to attach camera to
+	auto pos = Vector3Int::Zero;
+	GetJointAbsPosition(CamSlotId, &pos, CamMeshID);
+		//Get mesh2 to attach target to
+	auto pos2 = Vector3Int::Zero;
+	GetJointAbsPosition(TargetItem, &pos2, TargetMeshID);
+
+	if (
```
Comment (Sezzary) [inline TombEngine/Game/camera.cpp:372]: No need for `if (` to be on its own line.

File: TombEngine/Game/camera.cpp [hunk 5]
```diff
@@ -323,6 +332,131 @@ void MoveCamera(GameVector* ideal, int speed)
...
+		OldCam.target.y = Camera.target.y;
+		OldCam.target.z = Camera.target.z;
+		LastIdeal.x = pos.x;
+		LastIdeal.y = pos.y;
+		LastIdeal.z = pos.z;
+		LastIdeal.roomNumber = ideal->roomNumber;
+		LastTarget.x = pos2.x;
+		LastTarget.y = pos2.y;
+		LastTarget.z = pos2.z;
+			 }
```
Comment (Sezzary) [inline TombEngine/Game/camera.cpp:403]: The closing bracket should be lined up with the opening one above. :)

File: TombEngine/Game/camera.cpp [hunk 6]
```diff
@@ -323,6 +332,131 @@ void MoveCamera(GameVector* ideal, int speed)
...
+
+	if (LastAnglex - anglex > 100 || 
+		LastAngley - angley > 100 ||
+		LastAnglez - anglez > 100 
+		)
+		{
+		speed = 1;
+		}
+	else
+		{
```
Comment (Sezzary) [inline TombEngine/Game/camera.cpp:437]: TEN has a bit of a funny convention: where an `if` condition takes up one line _and_ the contents of its block also take up one line, we can leave out the brackets.

File: TombEngine/Game/camera.cpp [hunk 7]
```diff
@@ -1421,6 +1555,16 @@ void CalculateCamera()
...
 	}
 
+	
+
+	if (ItemCameraOn == true)
+	{
+		return;
+	}
+		
+
```
Comment (Sezzary) [inline TombEngine/Game/camera.cpp:1565]: Too many spaces. :)

**Review by Nemoel-Tomo — COMMENTED** [review]
fixed all known issues :)

**Review by Nemoel-Tomo — COMMENTED** [review]
hmm I hope the changes have been saved now

**Comment by Stranger1992** [comment]
How do you specify which camera is being attached? Or is it the game camera ?

**Comment by Nemoel-Tomo** [comment]
It is the game camera

**Pull request closed without merging**